### PR TITLE
Implemented claim function

### PR DESCRIPTION
**Pull request by amiromayer**
Now factory has claim function which checks whether a proxy contract is deployed for sender and deploys in case not and after claims required tokens

### CODE REVIEW

**Comment by amiromayer** [comment]
```npx waffle```

File: contracts/contracts/Factory.sol [hunk 1]
```diff
@@ -3,10 +3,28 @@ import "./CloneFactory.sol";
 import "./Storage.sol";
 
 interface ILinkdrop {
```
Comment (Dobrokhvalov) [inline contracts/contracts/Factory.sol]: why not to import interface from './interfaces/ILinkdrop' ?

File: contracts/contracts/Factory.sol [hunk 2]
```diff
@@ -17,20 +35,59 @@ contract Factory is Storage, CloneFactory {
...
     function deployProxy(address payable _sender) 
-    external 
+    public 
     returns (address payable) 
     {
 
         address payable proxy = createClone(implementation, keccak256(abi.encodePacked(_sender)));
-       
+
+        deployed[_sender] = proxy; // Added new
```
Comment (Dobrokhvalov) [inline contracts/contracts/Factory.sol]: don't get what's the meaning of the comment

File: contracts/test/Claim.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    receiverAddress = ethers.Wallet.createRandom().address
+    receiverSignature = await signReceiverAddress(link.linkKey, receiverAddress)
+
+    await expect(
+      factory.claim(
+        tokenAddress,
+        claimAmount,
+        expirationTime,
+        link.linkId,
+        sender.address, // New
```
Comment (Dobrokhvalov) [inline contracts/test/Claim.js]: don't get what's meaning of the comment

File: contracts/test/Claim.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
```
Comment (Dobrokhvalov) [inline contracts/test/Claim.js]: confusing comment, please remove

File: contracts/contracts/Factory.sol [hunk 5]
```diff
@@ -3,10 +3,28 @@ import "./CloneFactory.sol";
 import "./Storage.sol";
 
 interface ILinkdrop {
+
     function initializer(address payable _sender) external returns (bool);
+
+    function claim
```
Comment (Dobrokhvalov) [inline contracts/contracts/Factory.sol]: why not import interface from './interfaces/ILinkdrop' ?

Comment (Dobrokhvalov) [reply contracts/contracts/Factory.sol]: let's put initializer in the interface for now

Comment (amiromayer) [reply contracts/contracts/Factory.sol]: Sure

File: contracts/contracts/LinkdropERC721.sol [hunk 6]
```diff
@@ -2,30 +2,30 @@ pragma solidity >= 0.5.6;
...
-    bool initialized;
-
     function initializer
     (   
         address payable _sender
     ) 
     public
+    returns (bool)
     {
+        //require(msg.sender == owner, "Only owner has rights");
```
Comment (Dobrokhvalov) [inline contracts/contracts/LinkdropERC721.sol]: why this is added?

**Pull request merged**